NitroxPatcher/Patches/Dynamic/RangedAttackLastTarget_StartCasting_Patch.cs

```diff
@@ -0,0 +1,34 @@
+using System.Reflection;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Prevents non simulating players from running locally <see cref="RangedAttackLastTarget.StartCasting"/>.
+/// Broadcasts this event on the simulating player.
+/// </summary>
+public sealed partial class RangedAttackLastTarget_StartCasting_Patch : NitroxPatch, IDynamicPatch
+{
+    internal static readonly MethodInfo TARGET_METHOD = Reflect.Method((RangedAttackLastTarget t) => t.StartCasting(default));
+
+    public static void Prefix(RangedAttackLastTarget __instance)
+    {
+        if (!Resolve<AI>().IsCreatureWhitelisted(__instance.creature) ||
```

Comment:
This `if` is really hard to read.It should be splitted imo. The game won't go slower for few more IL lines generated